### PR TITLE
VM/SM/Blockchain/Trie/EVM: Copy() -> shallowCopy() function renaming

### PR DESCRIPTION
**Pull request by scorbajio**
TODO:
- [x] VM (see #2833)
- [x] StateManager
- [x] Blockchain
- [x] Trie
- [x] Evm
- [ ] Ethash

### CODE REVIEW

**Comment by holgerd77** [comment]
Quick note: after an exchange with @jochem-brouwer we realized that the EVM copy() method also has state and therefore should be renamed as well.

File: packages/client/lib/sync/fetcher/accountfetcher.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
+import { Fetcher, FetcherOptions } from './fetcher'
+import { Account } from '@ethereumjs/util'
+// import { Chain } from '../../blockchain'
```
Comment (gabrocheleau) [inline packages/client/lib/sync/fetcher/accountfetcher.ts:3]: can be removed?

Comment (scorbajio) [reply packages/client/lib/sync/fetcher/accountfetcher.ts:3]: yea I don't know how this crept it's way in. will remove.

**Comment by holgerd77** [comment]
Did a branch update here via UI

**Comment by holgerd77** [comment]
Another branch update via UI

**Review by holgerd77 — APPROVED** [review]
This looks pretty good, thanks a lot! 🤩

One thing: it would be good if we give this a mention in the direct code docs above the - now - `shallowCopy()` methods - what the "shallow" part of the copy actually is, so that this gets a bit of a description, something like the following for EVM:

```typescript
/**
* This method copies the EVM, current HF and EIP settings
* and returns a new EVM instance.
* 
* Note: this is only a shallow copy and both EVM instances
* will point to the same underlying state DB.
* 
* @returns EVMInterface
*/
```

Will merge though, this can be done in a small follow-up PR (also ok: alongside another PR).

**Comment by scorbajio** [comment]
Working on documentation update in #2855.

**Pull request merged**